products/redis/api.yaml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Google Cloud Memorystore for Redis
+prefix: gredis
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://redis.googleapis.com/v1beta1/
+scopes:
+   # TODO this is fake and made up but MM requires it
+  - https://www.googleapis.com/auth/redis
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Instance'
+    exclude: true
+    base_url: |
```

Comment:
A few things here:

1. The `base_url` is to provide a collection. You probably want `self_link_query` instead
2. Keep `base_url: projects/{{project}}/locations/{{region}}/instances`
3. Remove `self_link`

Reply:
Can you explain what a collection is in this context?

The reason I did it this way is that in the Terraform templates, we use `base_url` (via `collection_url`) to create the resource, and then `self_link` (via `self_link_url`) to read, update, and delete it. For redis, the query params are in the create call, so we would call POST to projects/{{project}}/locations/{{region}}/instances?instanceId={{name}} to create it. Then once we want to read it though, it's a GET to projects/{{project}}/locations/{{region}}/instances/{{name}}. The way it is now outputs the correct URLs for the Terraform templates- would you change it to use custom code, or is this fine?